xapian-letor/Makefile.am

```diff
@@ -115,9 +117,15 @@ include feature/Makefile.mk
 include ranker/Makefile.mk
 include scorer/Makefile.mk
 
-noinst_LTLIBRARIES += libcommon_str.la libgetopt.la
-libcommon_str_la_SOURCES = common/str.cc
-libgetopt_la_SOURCES = common/getopt.cc
+noinst_LTLIBRARIES += libcommon.la
+
+libcommon_la_SOURCES +=\
+	common/getopt.cc\
+	common/str.cc\
+	common/omassert.cc\
+	common/debuglog.cc\
+	common/errno_to_string.cc\
+	common/fileutils.cc
```

Comment:
I'm rather luke-warm on this part.

I don't think it's good to lump together "everything we use from common" and "GNU getopt (but only for systems where it isn't in libc)" like this.  Pretty much all the command line programs will use getopt, but probably most won't using debug logging or assertions.

Also, how come things build currently if these source files aren't currently getting linked in (except for common/str.cc)?

Reply:
@ojwb You are right we don't need any other sources other than str.cc and getopt.cc.I will fix it in a fixup commit.

Reply:
fixed in a fixup commit.